File: app/src/main/java/com/scop/org/minesweeper/generators/GridGenerator.java
package com.scop.org.minesweeper.generators;

import com.scop.org.minesweeper.elements.Grid;

public interface GridGenerator {
	void generateNewGrid(final Grid grid, int bombs, FinishCallback cb);

	@FunctionalInterface
	interface FinishCallback {
		void finished();
	}
}

File: app/src/main/java/org/oar/minesweeper/generators/GridGenerator.kt
package org.oar.minesweeper.generators

import org.oar.minesweeper.models.Grid
import org.oar.minesweeper.models.GridGenerationDetails
import org.oar.minesweeper.models.Tile

interface GridGenerator {
    fun generateNewGrid(grid: Grid, onFinish: (GridGenerationDetails) -> Unit)
    fun forceCleanSpot(grid: Grid, tile: Tile)
}